csharp/scripts/gen-assembly-info.py

```diff
@@ -1,34 +1,34 @@
 """
-Generates an `AssemblyInfo.cs` file that specifies the `AssemblyInformationalVersion` attribute.
-
-This attribute is set to the git version string of the repository."""
+Generates an `AssemblyInfo.cs` file that specifies a bunch of useful attributes
+that we want to set on our assemblies."""
 
 import pathlib
 import argparse
 
 
 def options():
     p = argparse.ArgumentParser(
-        description="Generate the assembly info file that contains the git SHA and branch name"
+        description="Generate an assembly info file."
     )
     p.add_argument("output", help="The path to the output file")
-    p.add_argument("gitinfo_files", nargs="+", help="The path to the gitinfo files")
+    p.add_argument("name", help="The name of the assembly")
     return p.parse_args()
 
 
 opts = options()
 
-gitfiles = dict()
-for file in map(pathlib.Path, opts.gitinfo_files):
-    gitfiles[file.name] = file
-
-version_string = gitfiles["git-ql-describe-all.log"].read_text().strip()
-version_string += f" ({gitfiles['git-ql-rev-parse.log'].read_text().strip()})"
-
 output_file = pathlib.Path(opts.output)
 output_file_contents = f"""
 using System.Reflection;
 
-[assembly: AssemblyInformationalVersion("{version_string}")]
+[assembly: AssemblyTitle("{opts.name}")]
+[assembly: AssemblyProduct("CodeQL")]
+[assembly: AssemblyVersion("1.0.0.0")]
+[assembly: AssemblyFileVersion("1.0.0.0")]
+[assembly: AssemblyCompany("GitHub")]
+[assembly: AssemblyCopyright("Copyright © 2024 GitHub")]
+
+[assembly: System.Runtime.Versioning.TargetFramework(".NETCoreApp,Version=v8.0", FrameworkDisplayName = ".NET 8.0")]
```

Comment:
Should we record in the C# wiki that this requires updating when we upgrade .NET?

Reply:
Good point. I'm doing it.